File: modules/cadastros.c
#include "../include/bd.h"
#include "../include/cadastros.h"

#include "../include/visual.h"
#include "../modules/produtos.c"
#include "../modules/entregas.c"
#include "../modules/cargas.c"
#include "../modules/vendas.c"

// ================= indicadores  ====================
int qtd_produtos(void) {
  return cursor_proximo_produto();
}

int qtd_entregas(void) {
  return 0;
}

int qtd_cargas(void) {
  return 0;
}

int qtd_vendas(void) {
  return 0;
}
//==========================================

void inserir_produtos(){
    Produto * novo_produto = alocar_produto();

  //==================== ler os campos do produto ====================
    printf(      "              ID: %03d\n", novo_produto->id);
    input_string("Nome do produto", novo_produto->nome, sizeof(novo_produto->nome));
    input_float(   "          Valor", &novo_produto->valor);
  //==================================================================

  //==================== validar =====================================
    validar_produto(novo_produto);
  //==================================================================

  //==================== inserir no banco de dados ===================
    inserirListaDupla(&bd()->produtos, novo_produto);
  //==================================================================
}

void buscar_produtos(const int id) {
    Produto chave = {0};
    chave.id = id;
    Produto *encontrado = pesquisarDuplo(bd()->produtos, &chave, localizar_produto);
    if (encontrado)
        printf("\nEncontrado: %s\n", encontrado->nome);
}

void listar_produtos() {
  imprimir_cabecalho_produto();
  imprimirDuplo(bd()->produtos, iprimir_produto);
  system("pause");
}

void inserir_entregas() {

}

void buscar_entregas() {

}

void listar_entregas() {

}

void inserir_cargas() {

}

void buscar_cargas() {

}
void listar_cargas() {

}

void inserir_vendas() {

}
void buscar_vendas() {

}
void listar_vendas() {

}

void relatorios(){
}

File: modules/cadastros.rs
use crate::visual::*;
use crate::bd::*;
use crate::modules::{produtos, entregas, cargas, vendas};

/// ================= indicadores  ====================
pub fn qtd_produtos() -> usize {
    bd().produtos.len()
}

pub fn qtd_entregas() -> usize {
    bd().entregas.len()
}

pub fn qtd_cargas() -> usize {
    bd().cargas.len()
}

pub fn qtd_vendas() -> usize {
    bd().vendas.len()
}
/// ====================================================

/// Inserir novo produto
pub fn inserir_produtos() {
    let mut novo_produto = produtos::Produto::novo();

    println!("              ID: {:03}", novo_produto.id);
    novo_produto.nome = input_string("Nome do produto");
    novo_produto.valor = input_float("Valor");

    produtos::validar_produto(&novo_produto);
    bd_mut().produtos.push(novo_produto);

    println!("✅ Produto inserido com sucesso!");
    pausar();
}

/// Buscar produto por ID
pub fn buscar_produtos(id: usize) {
    if let Some(produto) = bd().produtos.iter().find(|p| p.id == id) {
        println!("\nEncontrado: {}", produto.nome);
    } else {
        println!("\n❌ Produto não encontrado.");
    }
    pausar();
}

/// Listar produtos
pub fn listar_produtos() {
    produtos::imprimir_cabecalho_produto();
    for produto in &bd().produtos {
        produtos::imprimir_produto(produto);
    }
    pausar();
}

/// Entregas
pub fn inserir_entregas() {
    entregas::inserir_entrega();
}

pub fn buscar_entregas() {
    entregas::buscar_entrega();
}

pub fn listar_entregas() {
    entregas::listar_entregas();
}

/// Cargas
pub fn inserir_cargas() {
    cargas::inserir_carga();
}

pub fn buscar_cargas() {
    cargas::buscar_carga();
}

pub fn listar_cargas() {
    cargas::listar_cargas();
}

/// Vendas
pub fn inserir_vendas() {
    vendas::inserir_venda();
}

pub fn buscar_vendas() {
    vendas::buscar_venda();
}

pub fn listar_vendas() {
    vendas::listar_vendas();
}

/// Relatórios (placeholder)
pub fn relatorios() {
    println!("🔍 Módulo de relatórios ainda não implementado.");
    pausar();
}
